### PR TITLE
test(aggregates): cover ComputeAggregates

### LINKED ISSUE #601: test(aggregates): add unit tests for internal/aggregates
## Motivation

`internal/aggregates` has zero test files. Aggregate recompute is on the snapshot hot path — regressions silently corrupt dashboard data.

## Implementation information

- Add table-driven tests for aggregate recompute logic
- Cover edge cases: empty snapshots, single-snapshot, multi-owner aggregation
- Use real Postgres via testcontainers if pgx queries can't be cleanly mocked

## Supporting documentation

Part of umbrella #598

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

`internal/aggregates` had zero test files. `ComputeAggregates` is on the snapshot hot path — regressions here silently corrupt every dashboard reading. Closes #601.

## Implementation information

Table-driven tests for the pure `ComputeAggregates` math in `spec.go`:

- Empty values → single Shared zero row (with allocation `[]`, not `nil`)
- Single-owner asset + liability → assets, liabilities, net worth, allocation entry
- Multi-owner ordering: Shared (jointly owned) bucket first, then by id ascending
- Allocation entries sorted alphabetically by category
- Allocation sums multiple accounts in the same category
- Liability categories excluded from allocation
- Unknown account id (soft-deleted) → skipped, not panicked
- Known asset id → routed to Shared bucket
- Unknown asset id (not in `knownAssetIDs`) → skipped
- `firstOfMonth` normalization for January, mid-month, and December

No DB required — the function is pure. The issue suggested testcontainers; not needed since the math is decoupled from `store.go` (the store DB-side is exercised by the bb-tests).

## Supporting documentation

Part of umbrella #598.

> Changelog: test(aggregates): cover ComputeAggregates

### CODE REVIEW

File: backend-go/internal/aggregates/spec_test.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	if row.OwnerUserID != nil {
+		t.Errorf("OwnerUserID: want nil (Shared), got %v", *row.OwnerUserID)
+	}
+	if !row.TotalAssets.IsZero() || !row.TotalLiabilities.IsZero() || !row.NetWorth.IsZero() {
+		t.Errorf("want all zero, got assets=%s liab=%s nw=%s",
+			row.TotalAssets, row.TotalLiabilities, row.NetWorth)
+	}
+	if len(row.Allocation) != 0 {
+		t.Errorf("Allocation: want empty, got %v", row.Allocation)
+	}
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/aggregates/spec_test.go:39]: Added explicit nil-vs-empty assertion in fa35583.

File: backend-go/internal/aggregates/spec_test.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+func TestComputeAggregates_SingleOwnerAssetAndLiability(t *testing.T) {
+	owner := 7
+	accounts := []AccountInput{
+		{ID: 1, OwnerUserID: &owner, Type: "asset", Category: "Bank"},
+		{ID: 2, OwnerUserID: &owner, Type: "liability", Category: "Mortgage"},
+	}
+	values := []SnapshotValueInput{
+		{AccountID: new(1), Value: dec("1000")},
+		{AccountID: new(2), Value: dec("400")},
+	}
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/aggregates/spec_test.go:55]: Go 1.26 (this module's version) extended `new` to accept any expression — `new(1)` returns a fresh `*int` pointing at 1. CI `go-tests` + `lint-go` are green, which would have failed if it didn't compile. Leaving as-is.
